mods/hunger/init.lua

```diff
@@ -0,0 +1,317 @@
+
+hunger = {}
+
+local hunger_players = {}
+
+HUNGER_TICK = 800		-- time in seconds after that 1 hunger point is taken
+HUNGER_HEALTH_TICK = 4		-- time in seconds after player gets healed/damaged
+HUNGER_MOVE_TICK = 0.5		-- time in seconds after the movement is checked
+
+HUNGER_EXHAUST_DIG = 3		-- exhaustion increased this value after digged node
+HUNGER_EXHAUST_PLACE = 1	-- exhaustion increased this value after placed
+HUNGER_EXHAUST_MOVE = 1.5	-- exhaustion increased this value if player movement detected
+HUNGER_EXHAUST_LVL = 160	-- at what exhaustion player saturation gets lowered
+
+HUNGER_HEAL = 1			-- number of HP player gets healed after HUNGER_HEALTH_TICK
+HUNGER_HEAL_LVL = 5		-- lower level of saturation needed to get healed
+HUNGER_STARVE = 1		-- number of HP player gets damaged by hunger after HUNGER_HEALTH_TICK
+HUNGER_STARVE_LVL = 3		-- level of staturation that causes starving
+
+HUNGER_MAX = 30			-- maximum level of saturation
+HUNGER_VISUAL_MAX = 20		-- hud bar extends only to 20
+
+
+local function hunger_read(player)
+	local inv = player:get_inventory()
+	if not inv then
+		return nil
+	end
+
+	-- itemstack storage is offest by 1 to make the math work
+	local hgp = inv:get_stack("hunger", 1):get_count()
+	if hgp == 0 then
+		hgp = 21
+		inv:set_stack("hunger", 1, ItemStack({name = ":", count = hgp}))
+	end
+
+	return math.min(HUNGER_MAX + 1, hgp) - 1
+end
+
+local function hunger_save(player)
+	local inv = player:get_inventory()
+	if not inv then
+		return nil
+	end
+	local name = player:get_player_name()
+	local level = hunger_players[name].level
+
+	level = math.min(HUNGER_MAX, level)
+	level = math.max(level, 0)
+
+	inv:set_stack("hunger", 1, ItemStack({name = ":", count = level + 1}))
+	return true
+end
```

Comment:
Well, I know that this "method" works, but I'm stongly against using this hack as "official way" to save the hunger.
Thats one of the reasons why I open the PR for the engine.